cli/internal/cloudcmd/create.go

```diff
@@ -215,14 +220,84 @@ func (c *Creator) createAzure(ctx context.Context, cl terraformClient, config *c
 		return clusterid.File{}, err
 	}
 
+	// Patch the attestation policy to allow the cluster to boot while having secure boot disabled.
+	if err := patcher.Patch(ctx, tfOutput.AttestationURL, constants.EncodedAzureMAAPolicy); err != nil {
+		return clusterid.File{}, err
+	}
+
 	return clusterid.File{
-		CloudProvider: cloudprovider.Azure,
-		IP:            tfOutput.IP,
-		InitSecret:    []byte(tfOutput.Secret),
-		UID:           tfOutput.UID,
+		CloudProvider:  cloudprovider.Azure,
+		IP:             tfOutput.IP,
+		InitSecret:     []byte(tfOutput.Secret),
+		UID:            tfOutput.UID,
+		AttestationURL: tfOutput.AttestationURL,
 	}, nil
 }
 
+// PolicyPatcher interacts with Azure to update the attestation policy.
+type PolicyPatcher interface {
+	Patch(ctx context.Context, attestationURL, policy string) error
+}
+
+// NewPolicyPatcher returns a new PolicyPatcher.
+func NewPolicyPatcher() PolicyPatcher {
+	return policyPatcher{}
+}
+
+type policyPatcher struct{}
+
+// Patch updates the attestation policy to the base64-encoded attestation policy JWT for the given attestation URL.
+// https://learn.microsoft.com/en-us/azure/attestation/author-sign-policy#next-steps
+func (p policyPatcher) Patch(ctx context.Context, attestationURL, policy string) error {
+	// hacky way to update the MAA attestation policy. This should be changed as soon as either the Terraform provider supports it
+	// or the Go SDK gets updated to a recent API version.
+	// https://github.com/hashicorp/terraform-provider-azurerm/issues/20804
+	cred, err := azidentity.NewDefaultAzureCredential(nil)
+	if err != nil {
+		return fmt.Errorf("failed to create default Azure credentials: %w", err)
+	}
+	token, err := cred.GetToken(context.Background(), azpolicy.TokenRequestOptions{
+		Scopes: []string{"https://attest.azure.net/.default"},
+	})
+	if err != nil {
+		return fmt.Errorf("failed to create default Azure credentials: %w", err)
+	}
+
+	client := attestation.NewPolicyClient()
+
+	// azureGuest is the id for the "Azure VM" attestation type. Other types are documented here:
+	// https://learn.microsoft.com/en-us/rest/api/attestation/policy/set
+	req, err := client.SetPreparer(context.Background(), attestationURL, "azureGuest", p.encodeAttestationPolicy(constants.EncodedAzureMAAPolicy))
+	req.Header.Set("Authorization", "Bearer "+token.Token)
+	if err != nil {
+		return fmt.Errorf("failed to create request: %w", err)
+	}
+
+	res, err := client.Send(req)
+	if err != nil {
+		return fmt.Errorf("failed to send request: %w", err)
+	}
+	defer res.Body.Close()
+
+	if res.StatusCode != http.StatusOK {
+		return fmt.Errorf("failed to update attestation policy: %s", res.Status)
```

Comment:
```suggestion
	resp, err := client.Send(req)
	if err != nil {
		return fmt.Errorf("failed to send request: %w", err)
	}
	resp.Body.Close()

	if resp.StatusCode != http.StatusOK {
		return fmt.Errorf("failed to update attestation policy: %s", resp.Status)
```
This is usually called resp.
You can immediately close the Body if you don't need it.

Reply:
Out of interest: Has closing the body immediately any benefits (except for potential memory being freed)? Wouldn't it be more convenient for other developers who may extend the function (i.e. by further processing of the body) to keep it deferred? Don't take this as criticism to your review please, I really just want to extend my knowledge on this :)

Reply:
IMO it also makes the intent of the code clearer. I can see at once that Body isn't processed on purpose. Anyway, this is just a nit (forgot to prefix "nit: ") and you can choose what you prefer.